relay_handler.js

```diff
@@ -165,21 +167,31 @@ LazyRelayInReq.prototype.initRead =
 function initRead() {
     var self = this;
 
+    // TODO: wrap errors in protocol read errors?
+
     var res = self.reqFrame.bodyRW.lazy.readTTL(self.reqFrame);
     if (res.err) {
-        // TODO: wrap? protocol read error?
         return res.err;
     }
     self.timeout = res.value;
 
     res = self.reqFrame.bodyRW.lazy.readService(self.reqFrame);
     if (res.err) {
-        // TODO: wrap? protocol read error?
         return res.err;
     }
     self.serviceName = res.value;
 
-    // TODO: lazy read self.callerName
+    res = self.reqFrame.bodyRW.lazy.readHeaders(self.reqFrame);
+    if (res.err) {
+        return res.err;
+    }
+    self.headers = res.value;
```

Comment:
good catch, but that's not how this headers object works, see lazyRead in v2/header for more; fix pushed to this branch to actually use `self.headers.getValue(Buffer('cn'))`